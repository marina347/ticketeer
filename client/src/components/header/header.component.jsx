import React from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  selectCurrentUser,
  selectToken,
} from "../../redux/user/user.selectors";
import { ReactComponent as TicketeerLogo } from "../../assets/svg/ticket.svg";

import "./header.styles.scss";
import { closeSocket, createSocket } from "../../utils/client-socket";
import SideMenu from "../side-menu/side-menu.component";

export class HeaderComponent extends React.Component {
  componentDidMount() {
    createSocket(this.props.token);
  }
  componentWillUnmount() {
    closeSocket();
  }
  render() {
    const { history, currentUser } = this.props;

    return (
      <div id="header" className="header">
        <div id="header_logo_box" className="header__logo-box" onClick={() => history.push("/home")}>
          <TicketeerLogo className="header__logo" />
        </div>
        {currentUser ? (
          <SideMenu id="side_menu" currentUser={currentUser} history={history} />
        ) : null}
      </div>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser,
  token: selectToken,
});

const Header = connect(mapStateToProps)(withRouter(HeaderComponent));
export { Header };
