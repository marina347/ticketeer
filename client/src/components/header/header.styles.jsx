import styled from "styled-components";

export const HeaderContainer = styled.div`
  display: flex;
  flex-direction: row-reverse;
  padding: 15px;
`;
