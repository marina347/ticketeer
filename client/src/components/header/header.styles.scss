.header-container {
  padding: 2rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.logo-container {
}

.logo {
  height: 6.5rem;
  width: 6.5rem;
  opacity: 0.55;
}
