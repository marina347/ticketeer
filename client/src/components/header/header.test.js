import React from "react";
import { shallow } from "enzyme";
import { HeaderComponent } from "./header.component";

describe("Header component", () => {
  let wrapper;
  let mockProps;
  let currentUser;
  let history;
  let mockPush;
  beforeEach(() => {
    mockPush = jest.fn();
    currentUser = {
      name: "John",
    };
    history = {
      push: mockPush,
    };
    mockProps = {
      currentUser,
      history,
    };
    wrapper = shallow(<HeaderComponent {...mockProps} />);
  });

  it("Should render Header component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render only Logo if current user is null", () => {
    const newMockProps = {
      currentUser: null,
      history,
    };
    const newWrapper = shallow(<HeaderComponent {...newMockProps} />);
    expect(
      newWrapper.find('[id="header"]').find('[id="header_logo_box"]')
    ).not.toBeNull();
    expect(
      newWrapper.find('[id="header"]').find('[id="side_menu"]').exists()
    ).toBe(false);
  });

  it("Should render Logo and SideMenu if current user is not null", () => {
    expect(
      wrapper.find('[id="header"]').find('[id="header_logo_box"]')
    ).not.toBeNull();
    expect(
      wrapper.find('[id="header"]').find('[id="side_menu"]')
    ).not.toBeNull();
  });

  it("Should navigate to home page if logo is clicked", () => {
    wrapper
      .find('[id="header"]')
      .find('[id="header_logo_box"]')
      .simulate("click");
    expect(mockPush).toHaveBeenCalled();
  });
});
