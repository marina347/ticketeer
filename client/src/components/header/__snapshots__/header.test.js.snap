// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render Header component 1`] = `
<div
  className="header"
  id="header"
>
  <div
    className="header__logo-box"
    id="header_logo_box"
    onClick={[Function]}
  >
    <ForwardRef(SvgTicket)
      className="header__logo"
    />
  </div>
  <SideMenu
    currentUser={
      Object {
        "name": "John",
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    id="side_menu"
  />
</div>
`;
